lnbits/wallets/clnrest.py

```diff
@@ -0,0 +1,214 @@
+import asyncio
+import random
+from typing import AsyncGenerator, Dict, Optional
+
+import httpx
+from loguru import logger
+
+from lnbits import bolt11 as lnbits_bolt11
+from lnbits.settings import settings
+
+from ..core.models import Payment, PaymentStatus
+from .base import InvoiceResponse, PaymentResponse, StatusResponse, Wallet
+
+
+class CLNRestWallet(Wallet):
+    def __init__(self):
+        url = settings.cln_rest_url
+        macaroon = settings.cln_rest_macaroon
+        if not url or not macaroon:
+            raise Exception("cannot initialize CLN-rest")
+
+        # check if macaroon is base64 or hex
+        macaroon_ishex = True
+        try:
+            bytes.fromhex(macaroon)
+        except ValueError:
+            macaroon_ishex = False
+
+        self.url = url[:-1] if url.endswith("/") else url
+        self.auth = {
+            "macaroon": macaroon,
+            "accept": "application/json",
+        }
+        if macaroon_ishex:
+            self.auth["encodingtype"] = "hex"
+
+        self.cert = settings.cln_rest_cert or False
+
+    async def status(self) -> StatusResponse:
+        async with httpx.AsyncClient(verify=self.cert) as client:
```

Comment:
The httpx client should be reused just like the other wallets now do

Reply:
will do that!